plugins/Amplifier/Amplifier.cpp

```diff
@@ -50,99 +50,61 @@ Plugin::Descriptor PLUGIN_EXPORT amplifier_plugin_descriptor =
 }
 
 
-
-AmplifierEffect::AmplifierEffect( Model* parent, const Descriptor::SubPluginFeatures::Key* key ) :
-	Effect( &amplifier_plugin_descriptor, parent, key ),
-	m_ampControls( this )
+AmplifierEffect::AmplifierEffect(Model* parent, const Descriptor::SubPluginFeatures::Key* key) :
+	Effect(&amplifier_plugin_descriptor, parent, key),
+	m_ampControls(this)
 {
 }
 
 
-
-
-
-
-
-
-bool AmplifierEffect::processAudioBuffer( sampleFrame* buf, const fpp_t frames )
+bool AmplifierEffect::processAudioBuffer(sampleFrame* buf, const fpp_t frames)
 {
-	if( !isEnabled() || !isRunning () )
-	{
-		return( false );
-	}
+	if (!isEnabled() || !isRunning()) { return false ; }
 
 	double outSum = 0.0;
 	const float d = dryLevel();
 	const float w = wetLevel();
 
-	const ValueBuffer * volBuf = m_ampControls.m_volumeModel.valueBuffer();
-	const ValueBuffer * panBuf = m_ampControls.m_panModel.valueBuffer();
-	const ValueBuffer * leftBuf = m_ampControls.m_leftModel.valueBuffer();
-	const ValueBuffer * rightBuf = m_ampControls.m_rightModel.valueBuffer();
+	const ValueBuffer* volumeBuf = m_ampControls.m_volumeModel.valueBuffer();
+	const ValueBuffer* panBuf = m_ampControls.m_panModel.valueBuffer();
+	const ValueBuffer* leftBuf = m_ampControls.m_leftModel.valueBuffer();
+	const ValueBuffer* rightBuf = m_ampControls.m_rightModel.valueBuffer();
 
-	for( fpp_t f = 0; f < frames; ++f )
+	for (fpp_t f = 0; f < frames; ++f)
 	{
-//		qDebug( "offset %d, value %f", f, m_ampControls.m_volumeModel.value( f ) );
+		const float volume = (volumeBuf ? volumeBuf->value(f) : m_ampControls.m_volumeModel.value()) * 0.01f;
+		const float pan = (panBuf ? panBuf->value(f) : m_ampControls.m_panModel.value()) * 0.01f;
+		const float left = (leftBuf ? leftBuf->value(f) : m_ampControls.m_leftModel.value()) * 0.01f;
+		const float right = (rightBuf ? rightBuf->value(f) : m_ampControls.m_rightModel.value()) * 0.01f;
+
+		const float panLeft = std::min(1.0f, 1.0f - pan);
+		const float panRight = std::min(1.0f, 1.0f + pan);
 
 		auto s = std::array{buf[f][0], buf[f][1]};
 
-		// vol knob
-		if( volBuf )
-		{
-			s[0] *= volBuf->value( f ) * 0.01f;
-			s[1] *= volBuf->value( f ) * 0.01f;
-		}
-		else
-		{
-			s[0] *= m_ampControls.m_volumeModel.value() * 0.01f;
-			s[1] *= m_ampControls.m_volumeModel.value() * 0.01f;
-		}
-
-		// convert pan values to left/right values
-		const float pan = panBuf
-			? panBuf->value( f )
-			: m_ampControls.m_panModel.value();
-		const float left1 = pan <= 0
-			? 1.0
-			: 1.0 - pan * 0.01f;
-		const float right1 = pan >= 0
-			? 1.0
-			: 1.0 + pan * 0.01f;
-
-		// second stage amplification
-		const float left2 = leftBuf
-			? leftBuf->value( f )
-			: m_ampControls.m_leftModel.value();
-		const float right2 = rightBuf
-			? rightBuf->value( f )
-			: m_ampControls.m_rightModel.value();
-
-		s[0] *= left1 * left2 * 0.01;
-		s[1] *= right1 * right2 * 0.01;
+		s[0] *= volume * left * panLeft;
```

Comment:
I believe the array `s` could be removed and the whole calculation could be simplified to just:
```cpp
buf[f][0] *= d + w * volume * left * panLeft;
buf[f][1] *= d + w * volume * right * panRight;
```

Reply:
While that would be simpler, I think it's actually clearer how it currently is. I like that the wet value is calculated first, then the buffer set to a mixture of wet and dry. With this change, the factors for the wet value are split across both sides of the assignment, with the mixing code in between, and the wet value itself never materialises - you have to expand the product mentally to see how the code works.